interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/nodecompilation/FragmentParameterValidator.scala

```diff
@@ -62,8 +55,17 @@ object FragmentParameterValidator {
       fragmentParameter.name,
     )
 
-    val validationExpressionResult =
-      compileExpressionValidator(fragmentParameter, validationContext(fragmentParameter.name), expressionCompiler)
+    val validationExpressionResult = fragmentParameter.valueCompileTimeValidation
+      .map(validation =>
+        expressionCompiler
```

Comment:
I think that we don't need it anymore. Take a look that `FragmentCompleteDefinitionExtractor.extractParametersDefinition` is invoked always by `NodeCompiler`. We can leverage the fact, that parameters definition is "compiled" there, so we can just return `ValidationExpressionParameterValidatorToCompile`  in the definition. After that we can remove dependency to `ExpressionCompiler` and merge some `Fragment*DefinitionExtractor`s

Reply:
good idea